hoomd/mpcd/RejectionVirtualParticleFillerGPU.cuh

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2009-2019 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+// Maintainer: mphoward
+
+#ifndef MPCD_REJECTION_VIRTUAL_PARTICLE_FILLER_GPU_CUH_
+#define MPCD_REJECTION_VIRTUAL_PARTICLE_FILLER_GPU_CUH_
+
+/*!
+ * \file mpcd/RejectionVirtualParticleFillerGPU.cuh
+ * \brief Declaration of CUDA kernels for mpcd::RejectionVirtualParticleFillerGPU
+ */
+
+#include <cuda_runtime.h>
+
+#include "hoomd/HOOMDMath.h"
+#include "hoomd/BoxDim.h"
+#include "ParticleDataUtilities.h"
+
+#include "hoomd/RandomNumbers.h"
+#include "hoomd/RNGIdentifiers.h"
+
+
+namespace mpcd
+{
+namespace gpu
+{
+
+//! Common arguments passed to all geometry filling kernels
+struct draw_virtual_particles_args_t
+    {
+    //! Constructor
+    draw_virtual_particles_args_t(Scalar4 *_d_tmp_pos,
+                                  Scalar4 *_d_tmp_vel,
+                                  bool *_d_track_bounded_particles,
+                                  const Scalar3 _lo,
+                                  const Scalar3 _hi,
+                                  const unsigned int _first_tag,
+                                  const Scalar _vel_factor,
+                                  const unsigned int _filler_id,
+                                  const unsigned int _type,
+                                  const unsigned int _N_virt_max,
+                                  const unsigned int _timestep,
+                                  const unsigned int _seed,
+                                  const unsigned int _block_size)
+        : d_tmp_pos(_d_tmp_pos), d_tmp_vel(_d_tmp_vel), d_track_bounded_particles(_d_track_bounded_particles),
+        lo(_lo), hi(_hi), first_tag(_first_tag), vel_factor(_vel_factor), filler_id(_filler_id), type(_type),
+        N_virt_max(_N_virt_max), timestep(_timestep), seed(_seed), block_size(_block_size)
+        { }
+
+    Scalar4 *d_tmp_pos;
+    Scalar4 *d_tmp_vel;
+    bool *d_track_bounded_particles;
+    const Scalar3 lo;
+    const Scalar3 hi;
+    const unsigned int first_tag;
+    const Scalar vel_factor;
+    const unsigned int filler_id;
+    const unsigned int type;
+    const unsigned int N_virt_max;
+    const unsigned int timestep;
+    const unsigned int seed;
+    const unsigned int block_size;
+    };
+
+// Function declarations
+template<class Geometry>
+cudaError_t draw_virtual_particles(const draw_virtual_particles_args_t& args, const Geometry& geom);
+
+cudaError_t compact_indices(const bool *d_flags,
+                            const unsigned int num_items,
+                            unsigned int *d_out,
+                            unsigned int *d_num_selected_out,
+                            unsigned int *temp_storage);
+
+cudaError_t parallel_copy(unsigned int *d_compact_indices,
```

Comment:
Please use less generic names for these functions, as you could get clashes elsewhere in the code. For example:
`compact_virtual_particle_indices` and `copy_virtual_particles`